code/controllers/subsystem/processing/networks.dm

```diff
@@ -6,46 +6,57 @@ PROCESSING_SUBSYSTEM_DEF(networks)
 	flags = SS_KEEP_TIMING
 	init_order = INIT_ORDER_NETWORKS
 	var/datum/ntnet/station/station_network
-	var/assignment_hardware_id = HID_RESTRICTED_END
 	var/list/networks_by_id = list()				//id = network
-	var/list/interfaces_by_id = list()				//hardware id = component interface
-	var/resolve_collisions = TRUE
+	var/list/interfaces_by_hardware_id = list()
 
 /datum/controller/subsystem/processing/networks/Initialize()
 	station_network = new
 	station_network.register_map_supremecy()
 	. = ..()
 
-/datum/controller/subsystem/processing/networks/proc/register_network(datum/ntnet/network)
-	if(!networks_by_id[network.network_id])
-		networks_by_id[network.network_id] = network
-		return TRUE
-	return FALSE
+/datum/controller/subsystem/processing/networks/proc/register_network(network=null, datum/ntnet/parent=null)
+	var/datum/ntnet/net = null
+	if(!network)
+		net = station_network
+	else if(istext(network))
+		if(!networks_by_id[network])
+			if(!parent)
+				parent = station_network
+			net = new/datum/ntnet/(network, parent) // always a parrent to the main network
+			networks_by_id[network] = net
+			log_network("Network created for [parent] with the name '[network]'")
+	else if(istype(network, /datum/ntnet)) // custom network, like for sindies or charlie
+		net = network
+	if(!net)
+		log_runtime("No network created and not sure why?")
+	return net
 
-/datum/controller/subsystem/processing/networks/proc/unregister_network(datum/ntnet/network)
-	networks_by_id -= network.network_id
-	return TRUE
 
-/datum/controller/subsystem/processing/networks/proc/register_interface(datum/component/ntnet_interface/D)
-	if(!interfaces_by_id[D.hardware_id])
-		interfaces_by_id[D.hardware_id] = D
-		return TRUE
-	return FALSE
-
-/datum/controller/subsystem/processing/networks/proc/unregister_interface(datum/component/ntnet_interface/D)
-	interfaces_by_id -= D.hardware_id
-	return TRUE
 
+/// I think we should do this more like a routable ip address
 /datum/controller/subsystem/processing/networks/proc/get_next_HID()
-	var/string = "[num2text(assignment_hardware_id++, 12)]"
-	return make_address(string)
-
-/datum/controller/subsystem/processing/networks/proc/make_address(string)
-	if(!string)
-		return resolve_collisions? make_address("[num2text(rand(HID_RESTRICTED_END, 999999999), 12)]"):null
-	var/hex = md5(string)
-	if(!hex)
-		return		//errored
-	. = "[copytext_char(hex, 1, 9)]"		//16 ^ 8 possibilities I think.
-	if(interfaces_by_id[.])
-		return resolve_collisions? make_address("[num2text(rand(HID_RESTRICTED_END, 999999999), 12)]"):null
+	// lets get rid of recursion
+	var/static/list/collision_check = list()
```

Comment:
Might be better to have a global managed list, for better debugging.

Reply:
True, but kind of internaly debated about having a function handle its own checks on if a number has been used or not.